sway-lib-std/src/fixed_point.sw

```diff
@@ -0,0 +1,130 @@
+library fixed_point;
+//! A wrapper around U128 type for a library for Sway for mathematical functions operating with signed 64.64-bit fixed point numbers. 
+
+use ::u128::U128;
+use ::assert::assert;
+
+pub struct UFP64 {
+    value: U128 
+}
+
+impl core::ops::Eq for UFP64 {
+    pub fn eq(self, other: Self) -> bool {
+        self.value == other.value
+    }
+}
+
+impl core::ops::Ord for UFP64 {
+    pub fn gt(self, other: Self) -> bool {
+        self.value > other.value
+    }
+
+    pub fn lt(self, other: Self) -> bool {
+        self.value < other.value
+    }
+}
+
+impl core::ops::Add for UFP64 {
+    /// Add a UFP64 to a UFP64. Panics on overflow.
```

Comment:
Is reverting on an overflow desirable behaviour? cc @nfurfaro @adlerjohn

Reply:
Generally, yes. When this is not the desired behavior something like `overflowing_add()` must be used explicitly (or the vm flags must be modified to allow overflowing behavior).

Reply:
☝️ By default all arithmetic operations panic (or rather, revert) on overflow. One could add some logic to not revert if the panic-on-overflow flag is disabled, but that can be left for future work since it's not nearly as useful with floating point numbers.